src/struct_utils.cc

```diff
@@ -51,6 +51,40 @@ StructUtils::FindResult StructUtils::GetUInt64(const std::string& name,
   return OK;
 }
 
+StructUtils::FindResult StructUtils::GetBoolean(const std::string& name,
+                                                bool* value) {
+  const auto& fields = struct_pb_.fields();
+  const auto it = fields.find(name);
+  if (it == fields.end()) {
+    return MISSING;
+  }
+  if (it->second.kind_case() != google::protobuf::Value::kBoolValue) {
+    return WRONG_TYPE;
+  }
+  if (it->second.number_value() < 0) {
+    return NOT_POSITIVE;
+  }
+  *value = static_cast<bool>(it->second.bool_value());
+  return OK;
+}
+
+StructUtils::FindResult StructUtils::GetStruct(
+  const std::string& name, google::protobuf::Struct* value) {
+  const auto& fields = struct_pb_.fields();
+  const auto it = fields.find(name);
+  if (it == fields.end()) {
+    return MISSING;
+  }
+  if (it->second.kind_case() != google::protobuf::Value::kStructValue) {
+    return WRONG_TYPE;
+  }
+  if (it->second.number_value() < 0) {
+    return NOT_POSITIVE;
+  }
+  *value = static_cast<google::protobuf::Struct>(it->second.struct_value());
```

Comment:
This is a proto.Struct copy,  are you sure you need to copy it out?

Reply:
Removed static_cast.